src/reducers/reducer.js

```diff
@@ -1086,21 +1104,46 @@ export const onSetWCConnectionFailed = (state, { payload }) => ({
  * @param {Object} action.payload The network settings emitted in saga
  * @see updateNetworkSettings
  */
-export const onNetworkSettingsUpdate = (state) => ({
+export const onNetworkSettingsUpdateRequest = (state) => ({
   ...state,
   networkSettingsStatus: NETWORKSETTINGS_STATUS.LOADING,
 });
 
 /**
  * @param {Object} action.payload The network settings emitted in saga
- * @see updateNetworkSettings
+ * @see networkSettingsUpdateState customNetwork
+ */
+export const onNetworkSettingsUpdateState = (state, { payload }) => ({
+  ...state,
+  networkSettings: payload,
```

Comment:
Does it make sense to update the `networkSettingsStatus` to `NETWORKSETTINGS_STATUS.LOADING`?

Reply:
Here? No! Because this handling `onNetworkSettingsUpdateState` is executed in the context of wallet initialization only.